pandas/core/groupby/generic.py

```diff
@@ -1848,17 +1847,82 @@ def nunique(self, dropna: bool = True) -> DataFrame:
 
         return results
 
-    @doc(
-        _shared_docs["idxmax"],
-        numeric_only_default="False",
-    )
     def idxmax(
         self,
-        axis: Axis = 0,
+        axis: Axis = None,
```

Comment:
This is causing mypy to fail on the CI. Need to type-hint as `Axis | None`. Same for idxmin.

Reply:
Thanks, I have changed it